src/analysis/classic/warlock/demonology/modules/spells/Doomguard.tsx

```diff
@@ -0,0 +1,394 @@
+              Hurricane, then swap to your normal weapon before entering combat. It has only 15%
+              chance to happen, but it adds ~3.5% haste to your doomguard if you succeed!
+            </>
+          );
+        } else {
+          this.snapshotAdvice = (
+            <>
+              You failed to snapshot a buff. Try to summon your doomguard when all buffs are active.
+            </>
+          );
+        }
+        break;
+      default:
+        this.snapshotAdvice = (
+          <>
+            You failed to snapshot several buffs. Try to summon your doomguard when all buffs are
+            active.
+          </>
+        );
+        break;
+    }
+  }
+  private onDoomguardDamage(event: DamageEvent) {
+    if (this.doomguardSummonData.doomguard.totalDamage === undefined) {
+      this.doomguardSummonData.doomguard.totalDamage = 0;
+    }
+    this.doomguardSummonData.doomguard.totalDamage += event.amount;
+  }
+  private onFightEnd(event: FightEndEvent) {
+    if (
+      this.doomguardSummonData.doomguard.summonTimestamp !== 0 &&
+      this.doomguardSummonData.doomguard.deathTimestamp === 0
+    ) {
+      this.doomguardSummonData.doomguard.deathTimestamp = event.timestamp;
+    }
+  }
+  private onDoomguardDeath(event: DeathEvent) {
+    if (event.targetID === this.doomguardId) {
+      this.doomguardSummonData.doomguard.deathTimestamp = event.timestamp;
+      const doomguardDuration =
+        event.timestamp - this.doomguardSummonData.doomguard.summonTimestamp;
+      if (doomguardDuration >= 64000) {
+        this.doomguardSummonData.summonedDemonSummary = (
+          <li>
+            <PerformanceMark perf={QualitativePerformance.Perfect} /> You casted{' '}
+            <SpellLink spell={SPELLS.SUMMON_DOOMGUARD} /> at{' '}
+            {formatDuration(
+              this.doomguardSummonData.doomguard.summonTimestamp - this.owner.fight.start_time,
+            )}
+            , for a full <strong>{formatDuration(doomguardDuration)}</strong> duration and{' '}
+            <strong>{this.doomguardSummonData.doomguard.totalDamage}</strong> damage <br />
+          </li>
+        );
+      } else {
+        this.doomguardSummonData.summonedDemonSummary = (
+          <li>
+            <PerformanceMark perf={QualitativePerformance.Good} /> You casted{' '}
+            <SpellLink spell={SPELLS.SUMMON_DOOMGUARD} /> at{' '}
+            {formatDuration(
+              this.doomguardSummonData.doomguard.summonTimestamp - this.owner.fight.start_time,
+            )}
+            , for a <strong>{formatDuration(doomguardDuration)}</strong> duration. It did a total of{' '}
+            {this.doomguardSummonData.doomguard.totalDamage} damage, but your doomguard did not live
+            for its entire duration. Make sure that cast it soon enough so fight does not end before
+            your doomguard expires. <br />
+          </li>
```

Comment:
these duration/damage values are probably better as short lists or tables rather than prose, but i don't feel strongly about it.

Reply:
I have in mind to make it prettier in a near future, but I went to the essential with the time I have now ;)